kernel/userspace.c

```diff
@@ -317,23 +315,27 @@ void _k_object_wordlist_foreach(_wordlist_cb_func_t func, void *context)
 }
 #endif /* CONFIG_DYNAMIC_OBJECTS */
 
-static int thread_index_get(struct k_thread *t)
+static bool thread_index_get(struct k_thread *thread, u32_t *thread_index)
 {
 	struct _k_object *ko;
+	bool index_get_status = false;
 
-	ko = _k_object_find(t);
+	ko = _k_object_find(thread);
 
-	if (ko == NULL) {
-		return -1;
```

Comment:
Rule number?

Reply:
MISRA 15.5

Reply:
This is an advisory rule that has dozens of violations. We shouldn't fix this isolated here. My opinion is tackle the advisory rules after we deal with mandatory/required ones. And for this specific one, if we are going to fix we need to elaborate it more, so we can follow a pattern across the project.

Reply:
This was part of the bigger fix and not specific to 15.5. as mentioned before, inorder to correct the return value of thread_index_get() i did the changes.